tests/test_core.py

```diff
@@ -7629,6 +7629,33 @@ def test_wasm2js(self):
       with open('src.c.o.js.mem', 'rb') as f:
         self.assertTrue(f.read()[-1] != b'\0')
 
+  @no_fastcomp('wasm-backend specific feature')
+  def test_maybe_wasm2js(self):
+    if self.get_setting('WASM') == 0:
+      self.skipTest('redundant to test wasm2js in wasm2js* mode')
+    self.set_setting('MAYBE_WASM2JS', 1)
+    # see that running as wasm works
+    self.do_run_in_out_file_test('tests', 'core', 'test_hello_world')
+    # run wasm2js, bundle the code, and use the wasm2js path
+    cmd = [os.path.join(Building.get_binaryen_bin(), 'wasm2js'), '--emscripten', 'src.c.o.wasm']
+    if is_optimizing(self.emcc_args):
+      cmd += ['-O2']
+    js = run_process(cmd, stdout=PIPE).stdout
+    # assign the instantiate function to where it will be used
+    js = js.replace('function instantiate(asmLibraryArg, wasmMemory, wasmTable) {',
+                    "Module['__wasm2jsInstantiate__'] = function(asmLibraryArg, wasmMemory, wasmTable) {")
```

Comment:
This is more high-touch than I would like :( Could we clean this up in a followup by making wasm2js support this use case directly?